docs/guide/reactivity.md

```diff
@@ -226,6 +226,84 @@ The app that uses `Counter` uses the method `watch` to keep its progress bar syn
     ```{.textual path="docs/examples/guide/reactivity/dynamic_watch.py" press="enter,enter,enter"}
     ```
 
+## Recompose
+
+An alternative to a refresh is *recompose*.
+If you set `recompose=True` on a reactive, then Textual will remove all the child widgets and call [`compose()`][textual.widget.Widget.compose] again, when the reactive attribute changes.
+The process of removing and mounting new widgets occurs in a single update, so it will appear as though the content has simply updated.
+
+The following example uses recompose:
+
+=== "refresh03.py"
+
+    ```python hl_lines="10 12-13"
+    --8<-- "docs/examples/guide/reactivity/refresh03.py"
+    ```
+
+    1. Setting `recompose=True` will cause all child widgets to be removed and `compose` called again to add new widgets.
+    2. This `compose()` method will be called when `who` is changed.
+
+=== "refresh03.tcss"
+
+    ```css
+    --8<-- "docs/examples/guide/reactivity/refresh03.tcss"
+    ```
+
+=== "Output"
+
+    ```{.textual path="docs/examples/guide/reactivity/refresh03.py" press="P,a,u,l"}
+    ```
+
+While the end-result is identical to `refresh02.py`, this code works quite differently.
+The main difference is that recomposing creates an entirely new set of child widgets rather than updating existing widgets.
+So when the `who` attribute changes, the `Name` widget will replace its `Label` with a new instance (containing updated content).
+
+!!! warning 
+
+    You should avoid storing a reference to child widgets when using recompose.
+    Better to [query](../guide/queries.md) for a child widget when you need them.
+
+It is important to note that any child widgets will have their state reset after a recompose.
```

Comment:
This is disappointing - although clearly a hard problem to solve. I guess if you're very explicit about managing state then when the recompose happens you can load in the same data into a DataTable, reset the cursor position etc.

I wish we had a way of producing the new subtree and diffing it with the original so we could reuse nodes which have not had any data changed during the recompose.

Reply:
It may be possible if widgets had some kind of `replace_state` method, that could take a freshly constructed instance and copy the state.

But its not clear to me how to decided if a widget should be the same widget, or a new one.